### PR TITLE
fix(workspace): support resolving bare specifiers to npm pkgs within a workspace

### LINKED ISSUE #24605: Support bare specifiers for npm package members in a workspace
When not using byonm we don't support referring to other npm packages in the workspace via a bare specifier, but I think we should because it works in Node.

### PR DESCRIPTION
**Pull request by dsherret**
This makes bare specifiers for npm packages work when inside a workspace, which emulates the same behaviour as when there's a node_modules directory. The bare specifier can be overwritten by specifying an import map entry or package.json dependency entry.

* https://github.com/denoland/deno_config/pull/88

Closes #24605

### CODE REVIEW

File: tests/specs/npm/workspace_basic/b/main.ts [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import * as a1 from "@denotest/a";
 import * as a2 from "npm:@denotest/a@1";
-import * as a3 from "npm:@denotest/a@workspace";
```
Comment (dsherret) [inline tests/specs/npm/workspace_basic/b/main.ts:3]: https://github.com/denoland/deno_config/pull/87

File: cli/lsp/resolver.rs [hunk 2]
```diff
@@ -529,6 +529,17 @@ fn create_graph_resolver(
...
+        // todo(dsherret): we need a better way of determining the workspace
+        // root for this config data, but this is fine for now because this
+        // value is only used for determining if a specifier is in the workspace
+        // in order to resolve bare specifiers to workspace npm packages
+        npm_resolver
+          .and_then(|r| r.root_node_modules_path())
+          .and_then(|p| p.parent())
+          .and_then(|p| ModuleSpecifier::from_directory_path(p).ok())
+          .unwrap_or_else(|| ModuleSpecifier::parse("file:///").unwrap()),
+      ),
```
Comment (dsherret) [inline cli/lsp/resolver.rs:542]: @nayeemrmn is there a way to get the workspace root directory from `ConfigData`? I didn't see a way.

File: cli/lsp/resolver.rs
Comment (dsherret) [inline cli/lsp/resolver.rs:1]: The LSP doesn't support npm workspaces yet, so I did not implement this there (I'll add this PR to https://github.com/denoland/deno/issues/24505)

File: cli/lsp/resolver.rs (hunk 2 above)
Comment (nayeemrmn) [reply cli/lsp/resolver.rs:542]: You can add it as you see fit in `ConfigData::load_inner()`, it's just limited to what's used currently

Comment (dsherret) [reply cli/lsp/resolver.rs:538]: Thanks! Updated.

**Review by nayeemrmn — APPROVED** [review]
LGTM

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**